tools/toolchain/scripts/install_cosma.sh

```diff
@@ -0,0 +1,143 @@
+                           .. > cmake.log 2>&1
+                     ;;
+             esac
+                make -j $NPROCS > make.log 2>&1
+                install -d ${pkg_install_dir}/lib/cuda
+                [ -f libs/grid2grid/src/grid2grid/*.a ] && install -m 644 libs/grid2grid/src/grid2grid/*.a ${pkg_install_dir}/lib/cuda >> install.log 2>&1
+                [ -f libs/options/*.so ] && install -m 644 libs/options/*.so ${pkg_install_dir}/lib/cuda >> install.log 2>&1
+                install -m 644 src/cosma/*.a ${pkg_install_dir}/lib/cuda >> install.log 2>&1
+            fi
+        fi
+        COSMA_ROOT="${pkg_install_dir}"
+        COSMA_CFLAGS="-I'${pkg_install_dir}/include'"
+        COSMA_LDFLAGS="-L'${pkg_install_dir}/lib' -Wl,-rpath='${pkg_install_dir}/lib'"
+        COSMA_CUDA_LDFLAGS="-L'${pkg_install_dir}/lib/cuda' -Wl,-rpath='${pkg_install_dir}/lib/cuda'"
+        ;;
+    __SYSTEM__)
+        echo "==================== Finding cosma from system paths ===================="
+        check_command pkg-config --modversion cosma
+        add_include_from_paths COSMA_CFLAGS "cosma.h" $INCLUDE_PATHS
+        add_lib_from_paths COSMA_LDFLAGS "libcosma.*" $LIB_PATHS
+        ;;
+    __DONTUSE__)
+        ;;
+    *)
+        echo "==================== Linking cosma to user paths ===================="
+        pkg_install_dir="$with_spfft"
+        check_dir "$pkg_install_dir/lib"
+        check_dir "$pkg_install_dir/lib64"
+        check_dir "$pkg_install_dir/include"
+        COSMA_CFLAGS="-I'${pkg_install_dir}/include'"
+        COSMA_LDFLAGS="-L'${pkg_install_dir}/lib' -Wl,-rpath='${pkg_install_dir}/lib'"
+        ;;
+esac
+if [ "$with_cosma" != "__DONTUSE__" ] ; then
+    COSMA_LIBS="-lcosma_with_scalapack -lgrid2grid -lcosma -loptions"
+    if [ "$with_cosma" != "__SYSTEM__" ] ; then
+        cat << EOF > "${BUILDDIR}/setup_cosma"
+prepend_path LD_LIBRARY_PATH "$pkg_install_dir/lib64"
+prepend_path LD_RUN_PATH "$pkg_install_dir/lib64"
+prepend_path LIBRARY_PATH "$pkg_install_dir/lib64"
+prepend_path CPATH "$pkg_install_dir/include"
+export COSMA_INCLUDE_DIR="$pkg_install_dir/include"
+export COSMA_LIBS="-lcosma_with_scalapack -lgrid2grid -lcosma -loptions"
+export COSMA_ROOT="${pkg_install_dir}"
+export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:$pkg_install_dir/lib64/pkgconfig:$pkg_install_dir/lib/pkgconfig"
+EOF
+    fi
+    cat << EOF >> "${BUILDDIR}/setup_cosma"
+export COSMA_CFLAGS="${COSMA_CFLAGS}"
+export COSMA_LDFLAGS="${COSMA_LDFLAGS}"
+export COSMA_CUDA_LDFLAGS="${COSMA_CUDA_LDFLAGS}"
+export CP_DFLAGS="\${CP_DFLAGS} IF_MPI(IF_OMP(-D__COSMA|)|)"
+export CP_CFLAGS="\${CP_CFLAGS} ${COSMA_CFLAGS}"
+export CP_LDFLAGS="\${CP_LDFLAGS} IF_CUDA(${COSMA_CUDA_LDFLAGS}|${COSMA_LDFLAGS})"
+export COSMA_LIBRARY="-lcosma"
+export COSMA_ROOT="$pkg_install_dir"
+export COSMA_INCLUDE_DIR="$pkg_install_dir/include"
+export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:$pkg_install_dir/lib64/pkgconfig:$pkg_install_dir/lib/pkgconfig"
```

Comment:
`PKG_CONFIG_PATH` (and some other variables) seems to end up twice in there, correct?
And you once define a `COSMA_LIBRARY` but there is also a `COSMA_LIBS` with a slightly different set?

Reply:
please cleanup and/or clarify with a comment